kapre/backend.py

```diff
@@ -212,52 +212,26 @@ def mu_law_decoding(signal_mu, quantization_channels):
     return signal
 
 
-def freq_mask(self, input_f, param, name='freq_mask'):
-    """Apply masking to a spectrogram in the freq domain.
-    TensorFlow/io
-    
+def random_masking_along_axis(input_f, param, axis, name='masking'):
+    """
+    Apply masking to a spectrogram in the time/freq domain.
     Args:
       input: An audio spectogram.
       param: Parameter of freq masking.
       name: A name for the operation (optional).
-
     Returns:
       A tensor of spectrogram.
     """
     # TODO: Support audio with channel > 1.
-    freq_max = tf.shape(input_f)[1]
+    _max = tf.shape(input_f)[axis + 1]
+    _shape = [-1, 1, 1, 1]
```

Comment:
I don't think we need to reshape the input tensor. We can instead keep the code cleaner/shorter by just passing the right `axis` argument in the layer implementation. At the same time, this function would just blindly do masking over the specified `axis`.